R/Rules-methods.R

```diff
@@ -1725,6 +1725,92 @@ setMethod(
   }
 )
 
+## IncrementsMaxToxProb ----
+
+#' @describeIn maxDose determine the maximum possible next dose based on the
+#' probability of toxicity
+#' @param model (`GeneralModel`)\cr The model on which probabilities will be based
+#' @param `samples` (`Samples`)\cr The MCMC samples to which `model` will be applied
+#'
+#' @aliases maxDose-IncrementsMaxToxProb
+#'
+#' @export
+#' @example examples/Rules-method-maxDose-IncrementsMaxToxProb.R
+#'
+setMethod(
+  f = "maxDose",
+  signature = signature(
+    increments = "IncrementsMaxToxProb",
+    data = "DataOrdinal"
+  ),
+  definition = function(increments, data, model, samples, ...) {
+    assert_class(samples, "Samples")
+    assert_true(length(increments@prob) == length(data@yCategories) - 1)
+    nm <- names(data@yCategories)[2:length(data@yCategories)]
+    assert_set_equal(names(increments@prob), nm)
+
+    probs <- lapply(
+      seq_along(increments@prob),
+      function(g) {
+        fit(samples, model, data, grade = g, ...) %>%
+          dplyr::filter(middle < increments@prob[nm[g]]) %>%
+          utils::tail(1)
+      }
+    ) %>%
+      dplyr::bind_rows()
+    return(min(probs$dose))
```

Comment:
```suggestion
    min(probs$dose)
```
(avoid explicit return at end of function)